curlify.py

```diff
@@ -21,14 +31,26 @@ def to_curl(request, compressed=False, verify=True):
         ('-X', request.method),
     ]
 
-    for k, v in sorted(request.headers.items()):
-        parts += [('-H', '{0}: {1}'.format(k, v))]
+    for k, value in sorted(request.headers.items()):
+        parts += [('-H', '{0}: {1}'.format(k, value))]
 
     if request.body:
         body = request.body
+        data_arg = '-d'
         if isinstance(body, bytes):
-            body = body.decode('utf-8')
-        parts += [('-d', body)]
+            try:
+                body = body.decode('utf-8')
+            # handle binary files
+            except UnicodeDecodeError:
+                data_file = os.path.join(
+                    tempfile.tempdir,
+                    'curlify.{}.data'.format(md5(body).hexdigest())
+                )
+                with open(data_file, 'wb') as file:
+                    file.write(body)
```

Comment:
Using a tempfile in order to handle any size of data.
(See: `xargs --show-limits`)